src/peer_digest.cc

```diff
@@ -398,7 +398,7 @@ peerDigestHandleReply(void *data, StoreIOBuffer receivedData)
          */
         newsize = fetch->bufofs - retsize;
```

Comment:
This code is missing assertions, variable type changes, and a lot of other cleanup adjustments, but I think it is best to keep this bug-fixing PR in its minimal form, especially since I am not testing this code.